graphs/possible_bipartition.py

```diff
@@ -1,12 +1,36 @@
 # Can be used for BFS
-from collections import deque 
+from collections import deque
+
 
 def possible_bipartition(dislikes):
-    """ Will return True or False if the given graph
-        can be bipartitioned without neighboring nodes put
-        into the same partition.
-        Time Complexity: ?
-        Space Complexity: ?
+    """Will return True or False if the given graph
+    can be bipartitioned without neighboring nodes put
+    into the same partition.
+    Time Complexity: ?
+    Space Complexity: ?
     """
-    pass
 
+    if not dislikes:
+        return True
+
+    groups = [None] * len(dislikes)
+    group_1 = 0
+    group_2 = 1
+    start_dog = 0
+    while start_dog < len(dislikes) and not dislikes[start_dog]:
+        start_dog += 1
+
+    groups[start_dog] = group_1
+    queue = deque()
+    queue.append(start_dog)
+    while queue:
+        curr_dog = queue.popleft()
+        for disliked_dog in dislikes[curr_dog]:
+            if groups[disliked_dog] is None:
+                groups[disliked_dog] = (
+                    group_1 if groups[curr_dog] == group_1 else group_2
```

Comment:
🤔 I think you mixed up your logic a bit here. If the current dog is in `group_1` then you want the disliked dog to be in `group_2`.
```suggestion
                    group_2 if groups[curr_dog] == group_1 else group_1
```